cadc-vos-server/src/main/java/ca/nrc/cadc/vos/server/NodeDAO.java

```diff
@@ -1935,17 +1930,39 @@ protected String getSelectNodesByParentSQL(ContainerNode parent, Integer limit,
             sb.append(" WHERE parentID = ");
             sb.append(getNodeID(parent));
         }
-        else
+        else {
             sb.append(" WHERE parentID IS NULL");
-        if (withStart)
-            sb.append(" AND name >= ?");
+        }
+        
+        String sortColumn = "name";
+        if (sortProperty != null) {
+            switch (sortProperty.toString()) {
+                case VOS.PROPERTY_URI_DATE:
+                    sortColumn = "lastModified";
+                    break;
+                case VOS.PROPERTY_URI_CONTENTLENGTH:
+                    sortColumn = "contentLength";
+                    break;
+            }
```

Comment:
I think this case statement needs a default that throws... probably UnsupportedOperationException because we cannot sort on the requested property